packages/x-date-pickers-pro/src/DateRangeCalendar/DateRangeCalendar.tsx

```diff
@@ -558,7 +569,10 @@ const DateRangeCalendar = React.forwardRef(function DateRangeCalendar<TDate>(
               slots={slots}
               slotProps={slotProps}
             >
-              {utils.format(visibleMonths[month], 'monthAndYear')}
+              {utils.formatByString(
+                visibleMonths[month],
+                calendarHeaderProps.format ?? `${utils.formats.month} ${utils.formats.year}`,
```

Comment:
Do we want to allow for empty strings? From a quick look this could potentially throw an error if an invalid param gets passed into the formatter functions.

```suggestion
                calendarHeaderProps.format || `${utils.formats.month} ${utils.formats.year}`,
```

Reply:
For me if people pass an empty string it should use the empty format, not fallback to our format.
People can pass undefined if they want to conditionally override :+1:

Reply:
If it works that way, ok! I just haven't looked into each Adapter to seeing the lib functions support an empty string as parameter!

Reply:
On day.js is seems to format in ISO when you pass an empty string.
Tbh it's probably something people will never do, but I'm in favor of following the date library behavior, if people pass a format, we let the date library format with it even if it's a weird one.